cmake/thirdparty/SetupSeracThirdParty.cmake

```diff
@@ -240,9 +230,9 @@ if (NOT SERAC_THIRD_PARTY_LIBRARIES_FOUND)
         set(MFEM_CUSTOM_TARGET_PREFIX "mfem_" CACHE STRING "")
 
         # Tweaks needed after Spack converted to the HDF5 CMake build system
-        set(HDF5_TARGET_NAMES "hdf5::hdf5-static;hdf5::hdf5-shared" CACHE STRING "")
-        set(HDF5_IMPORT_CONFIG "RELWITHDEBINFO" CACHE STRING "")
-        set(HDF5_C_LIBRARY_hdf5_hl "hdf5::hdf5_hl-static" CACHE STRING "")
+        set(HDF5_TARGET_NAMES "hdf5-static;hdf5-shared" CACHE STRING "")
```

Comment:
This change seems like a step backwards. Are we sure these targets are now not namespaced?

Reply:
im not sure how to check this. i went to 

`/usr/WS2/smithdev/libs/serac/toss_4_x86_64_ib/2023_07_12_17_33_28/clang-14.0.6/hdf5-1.8.22-gu25dmytryol7p53glncizxl5jnxue3n/share/cmake/hdf5/hdf5-targets.cmake`

and saw :

```cmake
add_library(hdf5-static STATIC IMPORTED)
# and
add_library(hdf5_hl-static STATIC IMPORTED)
```

Reply:
That is very frustrating... hdf5@1.8.21 was namespaced... hdf5@1.8.22 is not.

It might be worth adding a check if the namespaced or non-namespaced version exists and just use what we find.